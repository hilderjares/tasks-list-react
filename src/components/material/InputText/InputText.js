import React from "react";

const TYPE = "text";

function InputText({ className, onChange, value, placeholder }) {
    return (
        <input
            type={TYPE}
            className={className}
            onChange={onChange}
            value={value}
            placeholder={placeholder}
            data-testid="input-text"
        />
    );
}

export default InputText;
