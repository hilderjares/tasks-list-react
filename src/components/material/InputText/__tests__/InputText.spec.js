import React from "react";
import {
    render,
    cleanup,
    fireEvent,
    waitForElement,
} from "@testing-library/react";
import InputText from "../InputText";

afterEach(cleanup);

describe("InputText component", () => {
    test("Matches the snapshot", () => {
        const { asFragment } = render(<InputText />);

        expect(asFragment().firstChild).toMatchSnapshot();
    });

    test("Should setter text 'new task' in input text", async () => {
        const { getByTestId } = render(<InputText />);

        const input = await waitForElement(() => getByTestId("input-text"));

        fireEvent.change(input, { target: { value: "new task" } });

        expect(input.value).toBe("new task");
    });

    test("Should not is empty", async () => {
        const { getByTestId } = render(<InputText />);

        const input = await waitForElement(() => getByTestId("input-text"));

        fireEvent.change(input, { target: { value: "testing" } });

        expect(input.value).not.toBe("");
    });
});
