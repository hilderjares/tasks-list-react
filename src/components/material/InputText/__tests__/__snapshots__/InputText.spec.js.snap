// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText component Matches the snapshot 1`] = `
<input
  data-testid="input-text"
  type="text"
  value=""
/>
`;
